### DISCUSSION TITLE
(git) posh-git package parameter?

### DISCUSSION THREAD

**Discussion by dahlbyk** [chocolatey-community/chocolatey-packages#1809] [Ideas]
- The [`poshgit` package](https://community.chocolatey.org/packages/poshgit) has been waiting for [some improvements](https://github.com/dahlbyk/posh-git/labels/Chocolatey) before it gets the (year-old 😢) 1.0 release:
  https://github.com/dahlbyk/posh-git/issues/772
- Since then, @dscho has been working towards adding an option to install posh-git:
  https://github.com/git-for-windows/git/issues/1384
- That installation will use the PowerShell Gallery, which is the preferred source for PowerShell modules:
  https://www.powershellgallery.com/packages/posh-git

Once I'm able to address the blockers and Git for Windows learns a posh-git option, it seems the Chocolatey package should learn one of two options:
- `/NoPoshGit` with posh-git installed by default
- `/PoshGit` to include it

With Chocolatey primary serving a shell-focused community, I lean toward including by default. I would then ship a final 1.0 release of the posh-git Chocolatey package that continues to depend on Git and just logs a warning that the package has been deprecated in favor of the Git package parameter.

Thoughts?

**Comment by pauby** [comment]
I'm really not in favour of bundling `poshgit` with `git`. Git is one thing. PoshGit is another. One of the advantages of Chocolatey / a package manager is the modularity you get. So if I want `poshgit` then I install `poshgit`. If I don't want `poshgit` I don't want to have to download it.

That modularity also works in the scenario where I want `poshgit` and it goes off and installs `git` for me (as I need both to have them work). The opposite is not true.

I can't see any reason why the bundling is a better idea than  having `poshgit` be a standalone?
